# Childcare Affordability in the U.S. 

The goal of this Capstone is to look at the trends of childcare affordability in relationship to median household income. It is my suspicion that the rate of childcare is increasing faster than wages.  

We will be looking at data provided by U.S. Department of Labor's Women's Bureau. The data is provided through an excel document titled "National Database of Childcare Prices" and it covers data from 2008-2018. https://www.dol.gov/agencies/wb/topics/featured-childcare

Definitions for columns can be found: https://www.dol.gov/sites/dolgov/files/WB/media/NationalDatabaseofChildcarePricesTechnicalGuideFinal.pdf

Definitions:

Childcare centers: typically located in commercial buildings and serve multiple groups or classrooms of similarly aged children. 

Family childcare homes: typically for small groups of children in a residential building, such as a house, apartment, or condo unit. 

MHI: Median household income.

TotalPop: Total population.

Households: Number of household.

MCInfant: Aggregated weekly, full-time median price charged for Center-based Care for infants (i.e. aged 0 through 23 months).

MCToddler: Aggregated weekly, full-time median price charged for Center-based Care for toddlers (i.e. aged 24 through 35 months).

MCPreschool: Aggregated weekly, full-time median price charged for Center-based Care for preschoolers (i.e. aged 36 through 54 months).

MFCCInfant: Aggregated weekly, full-time median price charged for Family Childcare for infants (i.e. aged 0 through 23 months).

MFCCToddler: Aggregated weekly, full-time median price charged for Family Childcare for toddlers (i.e. aged 24 through 35 months)

MFCCPreschool: Aggregated weekly, full-time median price charged for Family Childcare for preschoolers (i.e. aged 36 through 54 months).

In [1]:
import pandas as pd

In [2]:
# Read the Excel file
df = pd.read_excel('nationaldatabaseofchildcareprices2.xlsx', sheet_name='nationaldatabaseofchildcare')

# Display the DataFrame
df.head()

State_Name     County_Name  StudyYear      MHI  MHI_2018  TotalPop  \
0    Alabama  Autauga County     2008.0  50837.0  58462.55   49744.0   
1    Alabama  Autauga County     2009.0  51463.0  60211.71   49584.0   
2    Alabama  Autauga County     2010.0  53255.0  61775.80   53155.0   
3    Alabama  Autauga County     2011.0  53899.0  60366.88   53944.0   
4    Alabama  Autauga County     2012.0  53773.0  59150.30   54590.0   

   Households  MCInfant  MCToddler  MCPreschool  MFCCInfant  MFCCToddler  \
0     18373.0    104.95     104.95        85.92       83.45        83.45   
1     18288.0    105.11     105.11        87.59       87.39        87.39   
2     19718.0    105.28     105.28        89.26       91.33        91.33   
3     19998.0    105.45     105.45        90.93       95.28        95.28   
4     19934.0    105.61     105.61        92.60       99.22        99.22   

   MFCCPreschool  
0          81.40  
1          85.68  
2          89.96  
3          94.25  
4          98.53

# Data Wrangling 
Most of the data was cleaned up prior using excel. We deleted columns that we did not need. We will be focusing on California.  

In [3]:
df_california = df[df['State_Name'] == 'California']

In [4]:
'''Looking and handling missing values'''
# Check for missing values
missing_values = df_california.isnull()

# Summarize missing values
total_missing_per_column = missing_values.sum()

# Print the total number of missing values for each column
print(total_missing_per_column)

State_Name       0
County_Name      0
StudyYear        0
MHI              0
MHI_2018         0
TotalPop         0
Households       0
MCInfant         0
MCToddler        0
MCPreschool      0
MFCCInfant       0
MFCCToddler      0
MFCCPreschool    0
dtype: int64


In [5]:
#looking at the data

df_california.head(10)

State_Name     County_Name  StudyYear      MHI  MHI_2018   TotalPop  \
2045  California  Alameda County     2008.0  70079.0  80590.85  1457169.0   
2046  California  Alameda County     2009.0  68863.0  80569.71  1457095.0   
2047  California  Alameda County     2010.0  69384.0  80485.44  1477980.0   
2048  California  Alameda County     2011.0  70821.0  79319.52  1494876.0   
2049  California  Alameda County     2012.0  71516.0  78667.60  1515136.0   
2050  California  Alameda County     2013.0  72112.0  77880.96  1535248.0   
2051  California  Alameda County     2014.0  73775.0  78201.50  1559308.0   
2052  California  Alameda County     2015.0  75619.0  80156.14  1584983.0   
2053  California  Alameda County     2016.0  79831.0  83822.55  1605217.0   
2054  California  Alameda County     2017.0  85743.0  88315.29  1629615.0   

      Households  MCInfant  MCToddler  MCPreschool  MFCCInfant  MFCCToddler  \
2045    518352.0    301.77     214.22       214.22      192.32       178.26   
2046    520096.0    312.76     234.23       234.23      199.98       178.56   
2047    532026.0    313.14     234.95       234.95      201.32       181.19   
2048    536160.0    313.53     235.68       235.68      202.66       183.83   
2049    539179.0    313.91     236.40       236.40      204.00       186.46   
2050    545071.0    311.92     231.57       231.57      204.10       191.18   
2051    551734.0    309.92     226.74       226.74      204.19       195.89   
2052    558907.0    343.86     250.57       250.57      228.05       214.69   
2053    564293.0    377.80     274.39       274.39      251.91       233.48   
2054    569070.0    385.50     289.78       289.78      271.10       245.36   

      MFCCPreschool  
2045         178.26  
2046         178.56  
2047         181.19  
2048         183.83  
2049         186.46  
2050         191.18  
2051         195.89  
2052         214.69  
2053         233.48  
2054         245.36

In [6]:
# removing column since we are only looking at California
column_name_to_remove = 'State_Name'
df_california = df_california.drop(column_name_to_remove, axis=1)
df_california.head()



County_Name  StudyYear      MHI  MHI_2018   TotalPop  Households  \
2045  Alameda County     2008.0  70079.0  80590.85  1457169.0    518352.0   
2046  Alameda County     2009.0  68863.0  80569.71  1457095.0    520096.0   
2047  Alameda County     2010.0  69384.0  80485.44  1477980.0    532026.0   
2048  Alameda County     2011.0  70821.0  79319.52  1494876.0    536160.0   
2049  Alameda County     2012.0  71516.0  78667.60  1515136.0    539179.0   

      MCInfant  MCToddler  MCPreschool  MFCCInfant  MFCCToddler  MFCCPreschool  
2045    301.77     214.22       214.22      192.32       178.26         178.26  
2046    312.76     234.23       234.23      199.98       178.56         178.56  
2047    313.14     234.95       234.95      201.32       181.19         181.19  
2048    313.53     235.68       235.68      202.66       183.83         183.83  
2049    313.91     236.40       236.40      204.00       186.46         186.46

In [7]:
# checking type as we should change StudyYear, TotalPop, and Households into int
df_california.dtypes

County_Name       object
StudyYear        float64
MHI              float64
MHI_2018         float64
TotalPop         float64
Households       float64
MCInfant         float64
MCToddler        float64
MCPreschool      float64
MFCCInfant       float64
MFCCToddler      float64
MFCCPreschool    float64
dtype: object

In [8]:
# updating the aforementioned columns to int
columns_to_convert = ['StudyYear', 'TotalPop', 'Households']
df_california[columns_to_convert] = df_california[columns_to_convert].astype(int)


In [9]:
# checking if above worked
df_california.dtypes

County_Name       object
StudyYear          int32
MHI              float64
MHI_2018         float64
TotalPop           int32
Households         int32
MCInfant         float64
MCToddler        float64
MCPreschool      float64
MFCCInfant       float64
MFCCToddler      float64
MFCCPreschool    float64
dtype: object

In [10]:
df_california.head()

County_Name  StudyYear      MHI  MHI_2018  TotalPop  Households  \
2045  Alameda County       2008  70079.0  80590.85   1457169      518352   
2046  Alameda County       2009  68863.0  80569.71   1457095      520096   
2047  Alameda County       2010  69384.0  80485.44   1477980      532026   
2048  Alameda County       2011  70821.0  79319.52   1494876      536160   
2049  Alameda County       2012  71516.0  78667.60   1515136      539179   

      MCInfant  MCToddler  MCPreschool  MFCCInfant  MFCCToddler  MFCCPreschool  
2045    301.77     214.22       214.22      192.32       178.26         178.26  
2046    312.76     234.23       234.23      199.98       178.56         178.56  
2047    313.14     234.95       234.95      201.32       181.19         181.19  
2048    313.53     235.68       235.68      202.66       183.83         183.83  
2049    313.91     236.40       236.40      204.00       186.46         186.46

In [11]:
# resetting index
df = df_california.reset_index(drop=True)
df.head(5)


County_Name  StudyYear      MHI  MHI_2018  TotalPop  Households  \
0  Alameda County       2008  70079.0  80590.85   1457169      518352   
1  Alameda County       2009  68863.0  80569.71   1457095      520096   
2  Alameda County       2010  69384.0  80485.44   1477980      532026   
3  Alameda County       2011  70821.0  79319.52   1494876      536160   
4  Alameda County       2012  71516.0  78667.60   1515136      539179   

   MCInfant  MCToddler  MCPreschool  MFCCInfant  MFCCToddler  MFCCPreschool  
0    301.77     214.22       214.22      192.32       178.26         178.26  
1    312.76     234.23       234.23      199.98       178.56         178.56  
2    313.14     234.95       234.95      201.32       181.19         181.19  
3    313.53     235.68       235.68      202.66       183.83         183.83  
4    313.91     236.40       236.40      204.00       186.46         186.46

Summary

The first step I took was to download the data from U.S. Department of Labor's Women's Bureau's website. The original data was provided in an excel document and contained data from all of the United States. There were 34,568 rows and 227 columns. I combed through an excel file and removed all but 13 columns which focused on the data I wanted to observe. 

Once that was complete, after looking at missing values as well as making this project more relevant to my location, I decided to focus on California. From then on, I continued by removing non-California data. Following that, I changed some types from float to int since they needed to be whole numbers. 

Now, we will be looking at the following data: County_Name, StudyYear, MHI, MHI_2018, TotalPop, Households, MCInfant, MCToddler, MCPreschool, MFCCInfant, MFCCToddler, MFCCPreschool. These are all defined above. 

Next, we will explore the data further and see how well wage growth and childcare costs are related. 